webapp/Connector/src/view/MabGrid.js

```diff
@@ -720,6 +726,18 @@ Ext.define('Connector.view.MabGrid', {
             return false;
         }
 
+        // Fix for Issue 39229: Lag opening report when clicking on mAb report buttons
+        // Immediately go to the report view before prepareMAbReportQueries step below instead of staying on the grid view
+        // This report view is a temporary one, and will be later removed
```

Comment:
Can Connector.getQueryService().prepareMAbReportQueries be moved to mabreportview itself so we don't need to do the temp view thing?

Reply:
Agreed! now done!